### DISCUSSION TITLE
MSSQL in Automations Broken After Update

### DISCUSSION THREAD

**Discussion by R2bEEaton** [Budibase/budibase#6785] [Help] [answered]
I just updated from 1.0.218 to 1.1.13 with `budi hosting --update` and my MSSQL connection no longer works. I've checked the connection settings, removed and re-added it, etc. I discovered this because none of my Automations have worked since the update. I've obviously checked the automations and they all look fine. I also tried reverting my app and that didn't work either.

Automation error:
![image](https://user-images.githubusercontent.com/34921506/179520576-76a84aa5-8767-4922-8e90-65912e93e5a3.png)

Data source view (this used to not show up with any rows before I removed and readded the source):
![image](https://user-images.githubusercontent.com/34921506/179520762-c47595eb-37dd-4afb-b707-fdc4627ca5cf.png)

Automation step:
![image](https://user-images.githubusercontent.com/34921506/179520952-0179f704-5174-4f11-9b29-ea554c88a743.png)

**Comment by melohagan** [comment]
Hey @R2bEEaton 

Have you also updated the app itself to the latest version?

**Reply by R2bEEaton** [reply]
I did, but have since reverted it to try to fix this. Interestingly another automation of mine isn't working that doesn't have to do with the MSSQL database at all. It is triggered by a button and isn't even showing up in the automations log.

**Reply by melohagan** [reply]
If the minor versions of your app and Budibase versions don't match, then you're probably going to run into a variety of issues.
E.g. both need to be **v1.1.x** or **v1.0.x**

**Reply by melohagan** [reply]
Another point - the create row step shows `Use values`. This needs to be set to `Use bindings`.

**Reply by R2bEEaton** [reply]
I should probably make a new discussion for this, but I just went into the other automation that wasn't working and ran a test with JSON for the trigger.

```
{
  "fields": {
    "TripNbr": "6133",
    "DriverID": "ro_ta_users_us_27480d5f8b544855b7d9bc7117578a19",
    "Comment": "7/18 test"
  }
}
```

In the browser console I am getting an uncaught promise error again that was a known bug on previous versions. I have a filter in a query rows step.
![image](https://user-images.githubusercontent.com/34921506/179524192-81642fbc-5eb0-4693-a92a-30c5c20e1aa6.png)

Is there any way for me to downgrade Budibase as a whole to v1.0.218 rather than just an individual app. I need to get this all back into a working state.

**Reply by R2bEEaton** [reply]
> If the minor versions of your app and Budibase versions don't match, then you're probably going to run into a variety of issues. E.g. both need to be **v1.1.x** or **v1.0.x**

They are now on the same version, same problem.

> Another point - the create row step shows Use values. This needs to be set to Use bindings.

Good catch, this is a result of me re-adding the data source. Changed it to bindings and still no luck, same error appears in Automation Log.

**Reply by melohagan** [reply]
Here's an example of how you can adjust the docker compose file to revert the version: https://github.com/Budibase/budibase/discussions/6353

When you use the Budibase CLI it should create a docker compose file that can be edited.

**Reply by R2bEEaton** [reply]
> Here's an example of how you can adjust the docker compose file to revert the version: #6353
> 
> When you use the Budibase CLI it should create a docker compose file that can be edited.

That was helpful, I reverted back to v1.0.218 and our app is up and running as normal. It would be nice to have that automation log that is present in v1.1. Does this seem to you like an update issue with BB or is it something on my end?

**Reply by melohagan** [reply]
I think it's most likely a few teething problems with the latest release on our end. There are a few open bugs being worked on at the moment.
Hopefully as they get fixed you'll be able to upgrade. Sorry for the inconvenience!

**Reply by R2bEEaton** [reply]
Not a problem, thanks for your help!

**Reply by mike12345567** [reply]
Hi @R2bEEaton - had a look into your issue and it is in fact a problem with the current release, we will have fix out soon.

**Reply by mmalessa80** [reply]
@melohagan I seem to be having the same issue with MySQL datasource. I get the same output from my create row automation as above that the DB was not found.

**Comment by mike12345567** [comment, marked as answer]
Hi @mmalessa80 and @R2bEEaton - a fix has been deployed now to the self host images and our Cloud environment, this should be working for you now.

**Reply by R2bEEaton** [reply]
Thanks - it is :)
